man/make_split_fun.Rd

```diff
@@ -0,0 +1,112 @@
+}
+\value{
+A function for use as a custom split function.
+}
+\description{
+Create a Custom Splitting Function
+}
+\details{
+Custom split functions can be thought of as (up to) 3 different types of manipulations of
+the splitting process
+\enumerate{
+\item Preprocessing of the incoming data to be split
+\item (Row-splitting only) Customization of the core mapping of incoming data to facets, and
+\item Postprocessing operations on the set of facets (groups) generated by the split.
+}
+
+This function provides an interface to create custom split functions by implementing
+and specifying sets of operations in each of those classes of customization independently.
+
+Preprocessing  functions (1),  must  accept:  \code{df}, \code{spl},  \code{vals},
+\code{labels},  and  can  optionally accept  \code{.spl_context}.  They  then
+manipulate  \code{df} (the  incoming data  for the  split) and  return a
+modified data.frame.  This modified  data.frame \emph{must}  contain all
+columns present in the incoming  data.frame, but can add columns if
+necessary (though we note that these  new columns cannot be used in
+the layout as split or analysis variables, because they will not be
+present when validity checking is done).
+
+The preprocessing component is useful for things such as manipulating factor
+levels, e.g., to trim unobserved ones or to reorder levels based on observed counts,
+etc.
+
+Customization of core splitting (2) is  currently only supported in row
+splits. Core  splitting  functions  override  the  fundamental
+splitting procedure,  and are only  necessary in rare  cases. These
+must  accept  \code{spl}, \code{df},  \code{vals},  \code{labels},  and can  optionally
+accept \code{.spl_context}. They must return a named list with elements,
+all of  the same  length, as follows:  - \code{datasplit}  (containing a
+list of data.frames), -  \code{values} containing values associated with
+the   facets,   which   must   be   \code{character}   or   \code{SplitValue}
+objects. These  values will  appear in the  paths of  the resulting
+table.  - \code{labels} containing  the character labels associated with
+\code{values}
+
+Postprocessing functions (3) must  accept the result of  the core split
+as their first  argument (which as of writing can  be anything), in
+addition  to  \code{spl},  and   \code{fulldf},  and  can  optionally  accept
+\code{.spl_context}. They must each return a modified version
+of the same structure specified above for core splitting.
+
+In both the pre- and  post-processing cases, multiple functions can
+be specified. When this happens,  they are applied sequentially, in
+the order they  appear in the list passed to  the relevant argument
+(\code{pre} and \code{post}, respectively).
+}
+\examples{
+mysplitfun <- make_split_fun(pre = list(drop_facet_levels),
+                             post = list(add_overall_facet("ALL", "All Arms")))
```

Comment:
can me make a second example please?

mysplitfun2, where the list include two functions?